cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -302,12 +308,11 @@ struct weak_ordering_comparator_impl {
  *
  * @tparam Nullate A cudf::nullate type describing whether to check for nulls.
  */
-template <typename Nullate>
-using less_comparator =
-  weak_ordering_comparator_impl<device_row_comparator<Nullate>, weak_ordering::LESS>;
+template <template <typename> class Comparator, typename Nullate>
+using less_comparator = weak_ordering_comparator_impl<Comparator<Nullate>, weak_ordering::LESS>;
```

Comment:
I think the template-template parameter part is unnecessary. All we really care about is a type that models a binary callable that returns a `weak_ordering` value.

```suggestion
template <typename Comparator>
using less_comparator = weak_ordering_comparator_impl<Comparator, weak_ordering::LESS>;
```

With C++20:
```
template <typename C>
concept weak_ordering_comparator = requires(C c) {
   { c(left_index_t, right_index_t) } -> std::same_as<weak_ordering>;
   { c(right_index_t, left_index_t) } -> std::same_as<weak_ordering>;
};

template <weak_ordering_comparator C>
using less_comparator = weak_ordering_comparator_impl<C, weak_ordering::LESS>;
```

Reply:
Already fixed this in 529e94487dd3f700930e0d45b9d0670467430607 ;) I appreciate the fast review feedback, though!